### ISSUE TITLE
Is there an api to get bucket/collection stats ?

### ISSUE THREAD

**Issue by msathis** [Kinto/kinto#360]
Stats like no of documents, how much memory it holds etc will be useful.

**Comment by leplatrem** [comment]
:+1: 

No there isn't, but it's a good idea.

Related #173

**Comment by Natim** [comment]
@msathis do you have an idea of how you would like to access this information?

You can already have the number of records in a particular collection doing a head request to it:

    http HEAD https://kinto-ota.dev.mozaws.net/v1/buckets/blocklists/collections/certificates/records

    ETag: "1450717104423"
    Last-Modified: Mon, 21 Dec 2015 16:58:24 GMT
    Total-Records: 10

**Comment by msathis** [comment]
@Natim 
Right now
GET https://kinto-ota.dev.mozaws.net/v1/buckets/mybucket --auth="test:test"
returns data, permissions. Additionally it can return one more property "stats" as well.

The properties i would like to see in stats object
1. collections_count - (Applicable only for buckets)
2. records_count - Total records. Not sure whether we need it for bucket stats too.
2. storage_size - In bytes.
3. last_write - timestamp of last write.
4. last_read - timestamp of last read.

**Comment by msathis** [comment]
And the *data* property in the response contains only *last_modified* property. It can also return *created_at* timestamp as well.

**Comment by Natim** [comment]
Thank you for this it is really interesting :+1:

**Comment by tarekziade** [comment]
> last_read - timestamp of last read. (Not sure about this).

I don't think that one is easy to implement. When you have a stack with several nodes, you would need to have a complex implementation to be able to have that info. What is your use case for this info ?

**Comment by Natim** [comment]
The proposition is to add on collection a readonly stats parameter such as permissions and data which contains: 

**For buckets**
```
{
    "collections_count": 5,
    "storage_size": "478",
    "last_write": 1467022464234,
    "created_at": 1466357063998
}
```

**For collections**

{
    "records_count": "114",
    "storage_size": "83",
    "last_write": 1467022464234,
    "created_at": 1466357063998
}
```

**Comment by tarekziade** [comment]
Here are the two items I suggest we add in the collection or bucket metadata (and not used for the signing)

1. items_count 
2. storage_size - In bytes.

Those two fields will be useful to implement a quota feature among other things. I don't think it should be added as a plugin. It sounds like a core feature that can be added without having to deal with plugin indirections, in particular if we have to deal with conflict resolutions at some point.

If we reach the consensus on how those two fields are stored we can do what @leplatrem suggested:

- On create, sum the bytes, on delete subtracts, on update sum the difference between old/new. 
- On create/update, check if the body size exceeds setting size_per_item

@msathis  What is the use case for the created_at field ?

**Comment by Natim** [comment]
It is already possible to get the `last_write` and we can use kinto-changes for that.
Also having the creation date can be achieve with the kinto-history plugin that let you have an full audit trail.

**Comment by leplatrem** [comment]
What's the status of this ? Does the built-in plugin `kinto.plugins.quotas` covers this already?

BTW created date was later reported in issue #679
